src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXToZHigh.td

```diff
@@ -637,12 +640,13 @@ def replaceONNXMatMulAddPattern1 : Pat<
         (ZHighStickOp $b, (GetMatMulBiasLayoutStringAttr (GetRank $x),
                                                          (GetRank $y)), (GetDefaultSaturation)),
         (GetZeroI64Attr), (GetZeroI64Attr))),
-  [(IsMatMulLegalForZDNN $m), (HasRankOf<2> $y), (HasRankOf<1> $b),
-   (HaveSameLastDimR2R1 $y, $b)], [],
+  [(HasRankOf<2> $x), (HasRankOf<2> $y), (HasRankOf<1> $b), (HaveSameLastDimR2R1 $y, $b),
```

Comment:
We didn't check rank of x because this pattern can cover both `unstacked` and `bcast23` cases.

Reply:
got it